packages/next/src/lib/mkcert.ts

```diff
@@ -53,10 +57,37 @@ async function downloadBinary() {
 
     Log.info(`Download response was successful, writing to disk`)
 
-    const arrayBuffer = await response.arrayBuffer()
-    const buffer = Buffer.from(arrayBuffer)
+    const binaryWriteStream = fs.createWriteStream(binaryPath)
+
+    await response.body.pipeTo(
+      new WritableStream({
+        write(chunk) {
+          const { promise, resolve, reject } = new DetachedPromise<void>()
+          binaryWriteStream.write(chunk, (error) => {
+            if (error) {
+              reject(error)
+              return
+            }
+
+            resolve()
+          })
+          return promise
+        },
```

Comment:
```suggestion
        write(chunk) {
          return new Promise((resolve, reject) => {
            binaryWriteStream.write(chunk, (error) => {
              if (error) {
                reject(error)
                return
              }

              resolve()
            });
          });
        },
```

I don't think you have to use DetachedPromise here.

Reply:
Changed in [`2b37026` (#63527)](https://github.com/vercel/next.js/pull/63527/commits/2b370268ccfd5e4f5c598f51cf58d2ccc18a2827)